packages/react/src/contexts/ClerkContextProvider.tsx

```diff
@@ -76,7 +76,15 @@ export function ClerkContextProvider(props: ClerkContextProvider): JSX.Element |
         <SessionContext.Provider value={sessionCtx}>
           <OrganizationContext.Provider value={organizationCtx}>
             <AuthContext.Provider value={authCtx}>
-              <UserContext.Provider value={userCtx}>{children}</UserContext.Provider>
+              {/**
+               * This ensures that nothing will be displayed until clerk is loaded
+               * If the app requires for the sync/link to be performed, clerkLoaded will be false
+               * and none the content of the page will not flicker and no other request will be made
+               *
+               * At this moment checking clerkLoaded to achieve this makes sense because
+               * sync/link fires always by clerk-js
+               */}
+              <UserContext.Provider value={userCtx}>{clerkLoaded && children}</UserContext.Provider>
```

Comment:
❓ @panteliselef please ignore this comment if I'm mistaken, but I think this means that we won't be rendering any of the user's app at all before Clerk loads. Is that correct?

If so, this is definitely something that we should not do, as ClerkProvider in almost all scenarios wraps the whole app, blocking rendering until the `/client` and `/environment` requests resolve. 

Please confirm the above and let's discuss alternatives if this is something we should be doing  :)

Reply:
@nikosdouvlis I know that theses changes raise some red flags.
Currently sync/link fires before `/client` and `/enviroment` request are fired